File: atvi/modelos/telefone.ts
import Prototipo from "../interfaces/prototipo"

export default class Telefone implements Prototipo {
    public ddd: string
    public numero: string

    public clonar(): Prototipo {
        const clone = new Telefone()
        clone.ddd = this.ddd
        clone.numero = this.numero
        return clone
    }
}

File: AtvV/BackEnd/src/main/java/com/joao/AtvV/models/Telefone.java
package com.joao.AtvV.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Telefone {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(nullable = false, length = 3)
    private String ddd;

    @Column(nullable = false, length = 9, unique = true)
    private String numero;

    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "cliente_id")
    private Cliente cliente;
}
